homeassistant/components/airthings_ble/__init__.py

```diff
@@ -0,0 +1,74 @@
+"""The Airthings BLE integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from airthings_ble import AirthingsBluetoothDeviceData
+
+from homeassistant.components import bluetooth
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Airthings BLE device from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+    address = entry.unique_id
+
+    elevation = hass.config.elevation
+    is_metric = hass.config.units.is_metric
+    assert address is not None
+
+    ble_device = bluetooth.async_ble_device_from_address(hass, address.upper())
```

Comment:
```suggestion
    ble_device = bluetooth.async_ble_device_from_address(hass, address)
```

upper isn't needed here.  If you copied it from switchbot its only because of legacy libraries that we had to mutate it to upper